sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -428,21 +428,26 @@ public void onCallFailed(Exception e) {
             final Class<T> documentType,
             final WriteOptions writeOptions) {
         final DefaultAppCenterFuture<Document<T>> result = new DefaultAppCenterFuture<>();
-        getTokenAndCallCosmosDbApi(
-                partition,
-                result,
-                new TokenExchangeServiceCallback() {
+        if (mNetworkStateHelper.isNetworkConnected()) {
+            getTokenAndCallCosmosDbApi(
+                    partition,
+                    result,
+                    new TokenExchangeServiceCallback() {
 
-                    @Override
-                    public void callCosmosDb(final TokenResult tokenResult) {
-                        callCosmosDbCreateOrUpdateApi(tokenResult, document, documentType, partition, documentId, writeOptions, result);
-                    }
+                        @Override
+                        public void callCosmosDb(final TokenResult tokenResult) {
+                            callCosmosDbCreateOrUpdateApi(tokenResult, document, documentType, partition, documentId, writeOptions, result);
```

Comment:
if create succeed, should create the document in local storage too? with operation NIL?

Reply:
or leave it to create when read?

Reply:
We should cache when created yes, let's do it in a separate PR as this one is reviewed.

Reply:
will deal in the next pr 🙈